*** WlzJava.h	2007-10-03 16:10:18.631506763 +0100
--- WlzJava.h.mod	2007-10-03 16:18:07.174026256 +0100
***************
*** 18,24 ****
  #include <string.h>
  #include <stdio.h>
  #include <WlzBnd.h>
! #include <jni.h>
  #include <WlzJavaArray1D.h>
  #include <WlzJavaArray2D.h>
  #include <WlzJavaArray3D.h>
--- 18,24 ----
  #include <string.h>
  #include <stdio.h>
  #include <WlzBnd.h>
! #include <JavaVM/jni.h>
  #include <WlzJavaArray1D.h>
  #include <WlzJavaArray2D.h>
  #include <WlzJavaArray3D.h>
