*** WlzC2Java.java	2007-10-01 17:16:11.851855398 +0100
--- WlzC2Java.java.mod	2007-10-01 17:15:53.002139375 +0100
***************
*** 7649,7656 ****
          jj_expentry[i] = jj_lasttokens[i];
        }
        boolean exists = false;
!       for (java.util.Enumeration enum = jj_expentries.elements(); enum.hasMoreElements();) {
!         int[] oldentry = (int[])(enum.nextElement());
          if (oldentry.length == jj_expentry.length) {
            exists = true;
            for (int i = 0; i < jj_expentry.length; i++) {
--- 7649,7656 ----
          jj_expentry[i] = jj_lasttokens[i];
        }
        boolean exists = false;
!       for (java.util.Enumeration enoom = jj_expentries.elements(); enoom.hasMoreElements();) {
!         int[] oldentry = (int[])(enoom.nextElement());
          if (oldentry.length == jj_expentry.length) {
            exists = true;
            for (int i = 0; i < jj_expentry.length; i++) {
